dsb-util/src/main/java/com/spectralogic/dsbrowser/util/FunctionalExtensions.kt

```diff
@@ -47,3 +47,16 @@ inline fun <T> T?.exists(block: (T) -> Any?): Any? {
         block.invoke(this)
     }
 }
+
+private inline fun <T> T?.populated(block: T.() -> Boolean): Boolean {
+    return if (this == null) {
+        false
+    } else {
+        block.invoke(this)
+    }
+}
+
+fun String?.populated(): Boolean = this.populated { !isNotEmpty() }
+fun Collection<*>?.populated(): Boolean = this.populated { isNotEmpty() }
+fun Iterable<*>?.populated(): Boolean = this.populated { count() > 0 }
```

Comment:
I would avoid adding this check.  The count method can be quite expensive.  It would be better to implement this by getting the iterator, and then calling .hasNext on it.

Both cases are not all the desirable though, since some Iterables can be consumed on the first all to .iterator.

Reply:
lets take it out for now?

Reply:
If it's not being used, then I think that's a good idea.